### PR TITLE
added warning and set default image for unreleased versions of helm for #3770

### PR DESCRIPTION
**Pull request by goshlanguage**
I setup something in cmd/helm/installer/options.go to use the canary image in the event that the BuildMetaData for helm was "unreleased" and that UseCanary wasn't set, and the tiller-image flag wasn't used.

I separately repeated this logic in cmd/helm/init.go for the sake of keeping warnings and other output together.

### CODE REVIEW

**Comment by RyanHartje** [comment]
This should resolve #3770

File: cmd/helm/init.go [hunk 1]
```diff
@@ -323,6 +324,13 @@ func (i *initCmd) run() error {
 		fmt.Fprintln(i.out, "Not installing Tiller due to 'client-only' flag having been set")
 	}
 
+	if version.BuildMetadata == "unreleased" && !i.opts.UseCanary && len(i.opts.ImageSpec) == 0 {
```
Comment (bacongobbler) [inline cmd/helm/init.go]: This should probably check if the client has requested for `--client-only` as well, as that would affect some of the branch logic here. For example, a user running `helm init --client-only` won't need the `--canary-image` flag, so the warning should not be displayed.

**Review by bacongobbler — CHANGES_REQUESTED** [review]

**Comment by RyanHartje** [comment]
Thanks! I overlooked that.

**Comment by jstoja** [comment]
Hey @bacongobbler after the last changes of @RyanHartje , it should be okay, what do you think?

**Comment by mattjmcnaughton** [comment]
Thanks for working on this @RyanHartje  ! This has bitten me a couple of times, so I'm happy to see it getting fixed :)

**Review by bacongobbler — APPROVED** [review]

**Comment by bacongobbler** [comment]
This PR was submitted prior to the DCO switch, so I'm merging this as @ryanhartje has signed off on the CLA. Thanks for the PR!

**Pull request merged**